unstable/lens/define-nested.scrbl

```diff
@@ -0,0 +1,23 @@
+#lang scribble/manual
+
+@(require lens/private/doc-util/main)
+
+@title{Lenses for nested data}
+
+@defmodule[unstable/lens/define-nested]
+
+@defform[(define-nested-lenses [base-id base-lens-expr] clause ...)
+         #:grammar ([clause [sub-id sub-lens-expr]])]{
+A shorthand for defining composed lenses for nested data structures.
+
+For example, if there is a @racket[top] struct containing a
+@racket[middle] struct, which contains an @racket[x] field and a
+@racket[y] field, a form like:
+@(racketblock
+  (define-nested-lenses [top-middle top-middle-lens]
+    [x middle-x-lens]
+    [y middle-y-lens]))
```

Comment:
I think it would be better if we used examples with actual structs and lenses. To keep it simple, maybe a list of positions?

```
@lenses-unstable-examples[
  (struct/lens position (x y) #:transparent)
  (define-nested-lens [second second-lens]
    [x position-x-lens]
    [y position-y-lens])
  (lens-view second-x-lens (list (posn 0 0) (posn 1 4) (posn -2 -3)))
]
```